lib/spack/docs/environments.rst

```diff
@@ -368,6 +368,34 @@ from being added again.  At the same time, a spec that already exists in the
 environment, but only as a dependency, will be added to the environment as a
 root spec without the ``--no-add`` option.
 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Developing Packages in a Spack Environment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
Can this feature be used in package definitions as well?  I'm assuming yes, but it would be good to mention that or add some documentation in the packaging guide. For me that is where I most want to use this. I have a dependency on some packages that don't have formal releases or tags.  We just work off one branch/version they have and I'd like to snapshot with the git hash it for our own releases via spack.

Reply:
Yes, it can be used anywhere that a package and version are parsed. This PR adds a reference to it in the `basic_usage` section to address that, do you think I need to add it somewhere else?

Reply:
From a packaging perspective, I would strongly discourage using this to specify a dependency. I might be convinced otherwise by a compelling use case, but my instinct would be not to accept a PR that uses this feature in a `depends_on` directive.

Reply:
No that's sufficient. You make several good points.